File: src/main/java/com/ddd/trainings/idea/domain/TrainingIdeaFactory.java
package com.ddd.trainings.idea.domain;

import org.jmolecules.ddd.annotation.Factory;

@Factory
public class TrainingIdeaFactory {

  public static TrainingIdea create(String trainerId, String name) {
    IdeaNameVO ideaName = IdeaNameVO.from(name);
    IdeaId ideaId = IdeaId.create();
    TrainerId trainerIdVO = TrainerId.from(trainerId);
    TrainingIdea idea = new TrainingIdea(ideaId, trainerIdVO, ideaName);
    return idea;
  }
}

File: src/Ddd/Trainings/Idea/Domain/TrainingIdeaFactory.cs
namespace Ddd.Trainings.Idea.Domain
{
    public static class TrainingIdeaFactory
    {
        public static TrainingIdea Create(string trainerId, string name)
        {
            IdeaNameVO ideaName = IdeaNameVO.From(name);
            IdeaId ideaId = IdeaId.Create();
            TrainerId trainerIdVO = TrainerId.From(trainerId);
            return new TrainingIdea(ideaId, trainerIdVO, ideaName, TrainingDurationVO.Empty());
        }
    }
} 